rest_framework_json_api/filters/filter.py

```diff
@@ -0,0 +1,135 @@
+import re
+import warnings
+
+from rest_framework.exceptions import ValidationError
+from rest_framework.settings import api_settings
+
+from rest_framework_json_api.utils import format_value
+
+# django-filter is an optional package. Generate a dummy class if it's missing.
+try:
+    from django_filters.rest_framework import DjangoFilterBackend
```

Comment:
Instead of try except here let's move this into its own directory/module following the same structure as django filter does.

In this example `django_filters/backends.py` would be a suitable place to put it (potentially add a import in `django_filters/__init__.py`. This way `try` `except` is not needed as when a user of DJA doesn't import this module no error will occur even when django-filters is not installed.

django-filters does it the same way with `rest_framework` and I have seen this as well in other library such as `graphene-django`.

Reply:
Thanks. Will look at the examples, although this sounds like I'm losing my consistent naming of all JSONAPI filters being importable from `.filters`.